### DISCUSSION TITLE
issue with [@types/pg]. it's impossible to pass mixed-type values.

### DISCUSSION THREAD

**Discussion by hakyonglee** [DefinitelyTyped/DefinitelyTyped#69248] [Issues with a @types package]
When upgrading from version 8.11.2 to 8.11.3, there was a change in the code as follows. 

version 8.11.2
```
export interface QueryConfig<I extends any[] = any[]> {
    name?: string | undefined;
    text: string;
    values?: I | undefined;
    types?: CustomTypesConfig | undefined;
}
```

version 8.11.3
```
export interface QueryConfig<I = any[]> {
    name?: string | undefined;
    text: string;
    values?: QueryConfigValues<I>;
    types?: CustomTypesConfig | undefined;
}
export type QueryConfigValues<T> = T extends Array<infer U> ? (U extends never ? never : T) : never
```

Due to this, it has become impossible to provide mixed-type values as in pool.query({text, values: [1, 'a', null]}).
Please address this issue.

[a comment by an automated account is not shown]

**Comment by ghost** [comment]
Will be fixed by PR #69053

**Comment by elielberra** [comment]
It seems to me that the PR [69053](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69053) didn't solve this issue. I think that the distributive conditional of the `QueryConfigValues` type is not being correctly handled. I have added a possible solution and a potential new pull request on the typescript playground I am leaving at the end. Please let me know if something was not clear enough. Thanks! https://tsplay.dev/WY1y3W
